packages/gestalt/src/TableSortableHeaderCell.js

```diff
@@ -0,0 +1,93 @@
+// @flow strict
+import * as React from 'react';
+import classnames from 'classnames';
+import Box from './Box.js';
+import Icon from './Icon.js';
+import styles from './Table.css';
+
+type Props = {|
+  children: React.Node,
+  colSpan?: number,
+  onSortChange: ({
+    event:
+      | SyntheticMouseEvent<HTMLTableCellElement>
+      | SyntheticKeyboardEvent<HTMLTableCellElement>,
+  }) => void,
+  rowSpan?: number,
+  scope?: 'col' | 'colgroup' | 'row' | 'rowgroup',
+  sortOrder: 'asc' | 'desc',
+  status: 'active' | 'inactive',
+|};
+
+const SPACE_CHAR_CODE = 32;
+const ENTER_CHAR_CODE = 13;
+
+export default function TableSortableHeaderCell(props: Props) {
+  const {
+    children,
+    colSpan,
+    scope,
+    rowSpan,
+    status,
+    sortOrder,
+    onSortChange,
+  } = props;
+
+  const [isFocused, setFocused] = React.useState(false);
+  const [isHovered, setHovered] = React.useState(false);
+
+  const handleKeyPress = (
```

Comment:
@PayalModi Could we wrap the header `<Box />` in a `<Touchable />` instead? That way:

- Focus styles would work better 
![image](https://user-images.githubusercontent.com/127199/82912038-6fffe580-9f21-11ea-947d-f88fc074c8d3.png)
- Only the title + icon would be clickable (similar to material UI table and other tables)
- We don't have to copy/paste the clickable / focus code from `Touchable` in this file.

```
<th
      className={classes}
      scope={scope || 'col'}
      colSpan={colSpan}
      rowSpan={rowSpan}
    >
      <Touchable {...}>
        {children}
       <Touchable />
```

We might even be able to use `<Table.HeaderCell />` and pass all of this as children:

```
<Table.HeaderCell
      scope={scope}
      colSpan={colSpan}
      rowSpan={rowSpan}
    >
      <Touchable {...}>
        {children}
       <Touchable />
```

Reply:
Thanks for suggesting - I update to use TableHeaderCell wrapped around a Touchable!

Reply:
<img width="879" alt="Screen Shot 2020-05-26 at 3 00 45 PM" src="https://user-images.githubusercontent.com/5513405/82955518-e3c2e200-9f63-11ea-8368-2c582314131c.png">
this is the new focus state

Reply:
@PayalModi Thank you - might be worthwhile to follow up (we can do it later) to set `fullWidth` to `false` on `Touchable` - that way it would not span the whole 100% width.

Reply:
Ok! If I create a box with fit-content around the Touchable with fullWidth=False, the clickable area will only surround the title + arrow - I can put up a PR for that